cmd/bootstrapper/download/client.go

```diff
@@ -0,0 +1,97 @@
+package download
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/Dynatrace/dynatrace-bootstrapper/pkg/configure/oneagent/ca"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/url"
+	"github.com/spf13/afero"
+)
+
+type InstallerBuilder func(fs afero.Fs, dtc dtclient.Client, props *url.Properties) installer.Installer
+type DTClientBuilder func(url, apiToken, paasToken string, opts ...dtclient.Option) (dtclient.Client, error)
+
+type Client struct {
+	newInstaller InstallerBuilder
```

Comment:
```suggestion
	installBuilder InstallerBuilder
```
IMO, struct attribute with `new` a bit confusing.

Reply:
or may be we can addd importAlias to `codemoduleInstaller codemodule/installer` and use
```
type CLient struct {
     installer InstallerFunc
     dtClient DTClientFunc

}
```
or I saw somewhere pattern like so:
```
type CLient struct {
     installer InstallerOption
     dtClient DTClientOption

}
``` 
bc on line 23 you already have `Option func(*Client)` which is good.

Reply:
in golang, constructors tend to be called `func New() Type` or `func NewType() Type` calling it "builder" sounds wrong to me 
- I know we do it elsewhere, that doesn't mean we should keep doing that
- Considering we have actual "builders", to me calling this a "builder" is just confusing

Also it reads nicer to say:
```go
	oneAgentInstaller := cl.newInstaller(
		fs,
		client,
		&props,
	)
```
than
```go
	oneAgentInstaller := cl. installBuilder(
		fs,
		client,
		&props,
	)
```
it would be different if it looked more like (this is a builder)
```go
	oneAgentInstaller := cl. installBuilder().WithFS(fs).WithDT(client).WithProps(props).Build()
```

to call it `installer` and be correct(so it is not a constuctor func, but an actual `installer.Installer`), then I would need to re-arch the whole thing, because then the caller of `download.New()` would need to create the `dtclient`, so the caller would need to define the `download.Config` and so on.
- IMO, the current way is nicer, because it compartmentalizes more, and reflects the relation between packages  more
  - `cmd/bootstrapper/download` is a sub package of `cmd/bootstrapper`, and at the end of the day, it is optional (only necessary if we want to download the CodeModule as a ZIP) 

I will define a `type NewFunc func(afero.Fs, dtclient.Client, *Properties) installer.Installer` in the `installer/url` package to make it more reusable.

Reply:
I agree with what you wrote here, the concern was more about struct which has attribute which starts with `new`.
but let's use current impl.